subprojects/docs/src/docs/userguide/dep-man/01-core-dependency-management/dependency_verification.adoc

```diff
@@ -0,0 +1,900 @@
+
+Alternatively, you might want to enable verification on CI servers but not on local machines.
+
+Gradle actually provides 3 different verification modes:
+
+- `strict`, which is the default.
+Verification fails _as early as possible_, in order to avoid the use of compromised dependencies during the build.
+- `lenient`, which will run the build even if there are verification failures.
+The verification errors will be displayed during the build without causing a build failure.
+- `off` when verification is totally ignored.
+
+All those modes can be activated on the CLI using the `--dependency-verification` flag, for example:
+
+
+----
+./gradlew --dependency-verification lenient build
+----
+
+Alternatively, you can set the `org.gradle.dependency.verification` system property, either on the CLI:
+
+
+----
+./gradlew -Dorg.gradle.dependency.verification=lenient build
+----
+
+or in a `gradle.properties` file:
+
+
+----
+systemProp.org.gradle.dependency.verification=lenient
+----
+
+[[sec:trusting-artifacts]]
+=== Trusting some particular artifacts
+
+You might want to trust some artifacts more than others.
+For example, it's legitimate to think that artifacts produced in your company and found in your internal repository only are safe, but you want to check every external component.
+
+[NOTE]
+====
+This is a typical _company policy_.
+In practice, **nothing** prevents your internal repository from being compromised, so it's a good idea to check your internal artifacts too!
+====
+
+For this purpose, Gradle offers a way to automatically trust some artifacts.
+You can trust all artifacts in a group by adding this to your configuration:
+
+[source,xml]
+----
+<?xml version="1.0" encoding="UTF-8"?>
+<verification-metadata>
+   <configuration>
+      <trusted-artifacts>
+         <trust group="com.mycompany"/>
+      </trusted-artifacts>
+   </configuration>
+</verification-metadata>
+----
+
+This means that all components which group is `com.mycompany` will automatically be trusted.
+Trusted means that Gradle will not perform any verification whatsoever.
+
+The `trust` element accepts those attributes:
+
+- `group`, the group of the artifact to trust
+- `name`, the name of the artifact to trust
+- `version`, the version of the artifact to trust
+- `file`, the name of the artifact _file_ to trust
+- `regex`, a boolean saying if the `group`, `name`, `version` and `file` attributes need to be interpreted as regular expressions (defaults to `false`)
+
+In the example above it means that the trusted artifacts would be artifacts in `com.mycompany` but not `com.mycompany.other`.
+To trust all artifacts in `com.mycompany` and all subgroups, you can use:
+
+[source,xml]
+----
+<?xml version="1.0" encoding="UTF-8"?>
+<verification-metadata>
+   <configuration>
+      <trusted-artifacts>
+         <trust group="^com[.]mycompany($|([.].*))" regex="true"/>
```

Comment:
How about adding a special case like `group="com.mycompany.**"`?
It would be much easier to configure and understand than a cryptic `($|([.].*))`.

An alternative option is to use "group with subgroups" matching always, then if someone would need to use strict matching they would use regexp like `^com.mycompany$`. That regexp is much easier to understand.

Reply:
I'd rather not have a special case. The regex can be as cryptic as you need, and for most people, just writing `com.mycompany.*` even if _incorrect_, would do the trick.

Reply:
https://maven.apache.org/guides/mini/guide-naming-conventions.html, so groupId can't contain `*`

The motivation here is as follows:
1) `com.mycompany.**` is very understandable, so it conveys the message that subgroups are included
2) resorting to a regex for a common task (according to Gradle's own configuration file) seems to be not that nice. Being able to use `regex=true` is nice, however, "with subgroups" seems to be a quite common task because both Central and JCenter are based on "group with subgroups" security model
3) `group="^com[.]mycompany($|([.].*))"` conveys the message "I'm a regex", and then people need to carefully read it before they understand what that regex does. Hopefully, they won't update/review the configuration file often, so every time will be like "Ok, there's a regex" :)
4) `com.mycompany.*` will work for positive cases, however, it opens a security issue. I'm not sure it is the right thing to provide an API that makes it so easy to fall into a trap :(
Of course, the issue is like low-low (at the end of the day, PGP key still must match), but having an unexpected feature in a security/reproducibility field does not seem right.

Reply:
I understand but again I think we don't _have to_ provide many options. A regex is the most generic tool and works well. If in practice people complain, we can decide later. Also remember that if you're really serious about security, grouping like that shouldn't be an option, regex or not: you should leverage the artifact level and declare everything explicitly there.

Reply:
>grouping like that shouldn't be an option

That's true. That's why I listed security as the last item. The primary driver is readability.